### DISCUSSION TITLE
Implementing an ember simple auth authenticator

### DISCUSSION THREAD

**Discussion by timorusky** [firebase/emberfire#632] [Q&A] [answered]
Hi, I have some backend endpoints that I use for authentication that - in the end - just proxy to the Firebase Auth REST API. I was planning to implement a custom simple auth authenticator that calls my own backend. However, when I looked through the emberfire source code, I found the following lines in `addon/authenticators/firebase.ts`:

```
authenticate() {
    return reject(new Error('Please authenticate directly via the Firebase SDK.'));
}
```

Are there any security concerns and reasons why you should use the Firebase js sdk directly (I did some research and obviously the sdk only calls some Google endpoints in the last instance) or is it ok to use the REST API for initial authentication?

**Comment by maximilianmeier** [comment, marked as answer]
I don't see any issues here. Previous versions of emberfire also used an ember simple auth authenticator. 
To take the auth token of firebase and authenticate against your own api is also a proper use case
